citrineos/citrineos-core
# (Release 1.4) "null" Type for Optional Properties Part 3: Fix Update Partial Model

⚠️ The fix belongs to Release 1.4. Need to change the target branch before merging.

## Changes
1. Change the update behavior: set null instead of undefined to update the column value to null.
2. Fix bugs

## Tests
Test to set null using MessageInfo use case.
1. create a MessageInfo by sending `NotifyDisplayMessages` <img width="932" alt="Screenshot 2024-07-26 at 3 44 49 PM" src="https://github.com/user-attachments/assets/32412f6f-d7ba-4414-aeb0-5ff19be23f60"> <img width="895" alt="Screenshot 2024-07-26 at 3 30 35 PM" src="https://github.com/user-attachments/assets/cd0e5429-55ec-4faa-a2eb-ac9d7361b62b">
2. update MessageInfo without property, `endDateTime` and expect it is set to null in db. <img width="933" alt="Screenshot 2024-07-26 at 3 46 40 PM" src="https://github.com/user-attachments/assets/47a8df52-d011-441c-84cb-b57bb469d9e1"> <img width="916" alt="Screenshot 2024-07-26 at 3 30 41 PM" src="https://github.com/user-attachments/assets/efc8e2da-33d1-4e09-a368-b9fcebeee74d">

## Related PRs
Part 1: https://github.com/citrineos/citrineos-core/pull/183
Part 2: https://github.com/citrineos/citrineos-core/pull/184
